go/CWE-078/CommandInjection.ql

```diff
@@ -0,0 +1,39 @@
+/**
+ * @name Command built from user-controlled sources
+ * @description Building a system command from user-controlled sources is vulnerable to insertion of
+ *              malicious code by the user.
+ * @kind path-problem
+ * @problem.severity error
+ * @security-severity 9.8
+ * @precision high
+ * @id go/command-injection
+ * @tags security
+ *       external/cwe/cwe-078
+ */
+
+import go
+import semmle.go.security.CommandInjection
+import DataFlow::PathGraph
+import semmle.go.security.FlowSources
+
+//Override CommandInjection::Configuration to use the in-use sources
+class InUseCommandInjectionConfiguration extends CommandInjection::Configuration {
+  override predicate isSource(DataFlow::Node node) {
+    exists(UntrustedFlowSource source, Function function, DataFlow::CallNode callNode |
+      source.asExpr() = node.asExpr() and
+
+      source.(DataFlow::ExprNode).asExpr().getEnclosingFunction() = function.getFuncDecl() and
```

Comment:
Add a third check here for public vs private ...  the unused code is might really used somehow, either by an external mechanism we can’t observe